src/commands/terraform_deploy.yml

```diff
@@ -29,12 +29,20 @@ steps:
         SCRIPT_ASSUME: <<include(scripts/aws-assume.sh)>>
         SCRIPT_SET_VARS: <<include(scripts/set-variables.sh)>>
       command: <<include(scripts/terraform-init.sh)>>
-  - attach_workspace:
-      at: /tmp/workspace
   - run:
-      name: Assess Plan Change
+      name: Terraform Plan
       environment:
-        ENV: "development"
+        AWS_ACCESS_KEY_ID: ""
+        AWS_SECRET_ACCESS_KEY: ""
+        DIR: <<parameters.dir>>
+        ENV: <<parameters.env>>
+        TF_VAR_aws_profile: "assumed-role"
+        TF_VAR_env: <<parameters.env>>
+      command: <<include(scripts/terraform-plan.sh)>>
+  - run:
+      name: Assess Plan Changed
+      environment:
+        CHANGE_FILE: /tmp/development-tfplan.change
```

Comment:
Takže při deployi se znovu vytvoří plán.

Reply:
Do `CHANGE_FILE` pro ten `cancel-job.sh` se předává cesta k souboru, který vygeneroval `scripts/terraform-plan.sh` a píše do něj jestli došlo k nějakým změnám v plánu (https://github.com/FigurePOS/circle-ci-terraform-orb/blob/f5d550dd556d724f03c6305b0b4581002d8c956e/src/scripts/terraform-plan.sh#L21-L23). `cancel-job.sh` se podle něj rozhoduje jestli zrušit job.

Musím to parametrizovat, protože `cancel-job.sh` používá i validace (`src/commands/terraform_validate.yml` o kousek níž) a ta ho má na jiné cestě, protože ho načte z workspace.

Reply:
Jestli to není srozumitelný, tak dej vědět. Já už se v tom asi sám ztrácím. 😅

Reply:
Smysl to dava, jako abych ti to prosel nejak do hloubky, tak bych si to musel vic nastudovat, ale na prvni pohled dobry :D

Reply:
V pohodě. Jsem i trochu přemýšlel jestli tady to ověřování jestli se plán změnil a cancelování jobu úplně nezrušit když se to workflow triggeruje jen na tu změnu v adresáři. By se to aspoň cleý docela zjednodušilo. Ale takhle je to zase takový chytřejší, no. 😄

Reply:
Za předpokladu že to bude fungovat správně. 😆 Hm, já to asi nakonec oddělám, je to celkem mess. 🤔 Nebo co myslíš?

Reply:
Jako ja jsem spis pro jednodussi a citelnejsi kod, pokud v tom navic nevidis velkej benefit, tak bych to uprimne asi oddelal. Kdyby to byla nejaka security dira nebo neco, tak to asi dava smysl, ale tady bych rekl, ze to neni uplne potreba 🤔

Reply:
jo, jsem taky pro 👍